test/widgets/board_test.dart

```diff
@@ -29,11 +36,16 @@ void main() {
     });
 
     testWidgets('cannot select piece', (WidgetTester tester) async {
+      reset(onTappedSquare);
```

Comment:
Better to use `setUp` (or `tearDown`) to reset the mock before each test, even if they don't use it.